integration/yarn/src/main/java/tachyon/yarn/Client.java

```diff
@@ -0,0 +1,378 @@
+    setupAppMasterEnv(appMasterEnv);
+    mAmContainer.setEnvironment(appMasterEnv);
+  }
+
+  private void setupAppMasterJar(LocalResource appMasterJar) throws IOException {
+    Path jarHdfsPath = new Path(mAppMasterJarHdfs); // known path to jar file on HDFS
+    FileStatus jarStat = FileSystem.get(mYarnConf).getFileStatus(jarHdfsPath);
+    appMasterJar.setResource(ConverterUtils.getYarnUrlFromPath(jarHdfsPath));
+    appMasterJar.setSize(jarStat.getLen());
+    appMasterJar.setTimestamp(jarStat.getModificationTime());
+    appMasterJar.setType(LocalResourceType.FILE);
+    appMasterJar.setVisibility(LocalResourceVisibility.PUBLIC);
+  }
+
+  private void setupAppMasterEnv(Map<String, String> appMasterEnv) {
+    for (String c : mYarnConf.getStrings(YarnConfiguration.YARN_APPLICATION_CLASSPATH,
+        YarnConfiguration.DEFAULT_YARN_APPLICATION_CLASSPATH)) {
+      Apps.addToEnvironment(appMasterEnv, ApplicationConstants.Environment.CLASSPATH.name(),
+          c.trim(), ApplicationConstants.CLASS_PATH_SEPARATOR);
+    }
+    Apps.addToEnvironment(appMasterEnv, ApplicationConstants.Environment.CLASSPATH.name(),
+        Environment.PWD.$() + File.separator + "*", ApplicationConstants.CLASS_PATH_SEPARATOR);
+  }
+
+  /**
+   * Sets up the application submission context.
+   */
+  private void setupApplicationSubmissionContext() {
+    // set the application name
+    mAppContext.setApplicationName(mAppName);
+
+    // Set up resource type requirements
+    // For now, both memory and vcores are supported, so we set memory and vcores requirements
+    Resource capability = Resource.newInstance(mAmMemoryInMB, mAmVCores);
+    mAppContext.setResource(capability);
+
+    // Set the queue to which this application is to be submitted in the RM
+    mAppContext.setQueue(mAmQueue);
+
+    // Set the AM container spec
+    mAppContext.setAMContainerSpec(mAmContainer);
+
+    // Set the priority for the application master
+    mAppContext.setPriority(Priority.newInstance(mAmPriority));
+  }
+
+  /**
+   * Monitors the submitted application for completion.
+   *
+   * @return true if application completed successfully
+   * @throws YarnException if errors occur when obtaining application report from ResourceManager
+   * @throws IOException if errors occur when obtaining application report from ResourceManager
+   */
+  private boolean monitorApplication() throws YarnException, IOException {
+    DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    while (true) {
+      // Check app status every 10 second.
+      CommonUtils.sleepMs(10 * Constants.SECOND_MS);
+      Date date = new Date();
+      String timeDateStr = dateFormat.format(date);
+
+      // Get application report for the appId we are interested in
+      ApplicationReport report = mYarnClient.getApplicationReport(mAppId);
+
+      System.out.println(timeDateStr + " Got application report from ASM for appId="
```

Comment:
I suspect people will run this using commandline, so using `System.out.println` to get message to console through this file.